doc/source/ray-air/examples/vicuna_13b_lightning_deepspeed_finetune.ipynb

```diff
@@ -0,0 +1,1468 @@
+    "    # Take a subset to accelerate release tests\n",
+    "    limit_train_batches=20,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, combine all the configurations with {class}`LightningConfigBuilder <ray.train.lightning.LightningConfigBuilder>` and instantiate a LightningTrainer. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ray.air.config import CheckpointConfig, RunConfig, ScalingConfig\n",
+    "\n",
+    "trainer = LightningTrainer(\n",
+    "    lightning_config=lightning_config.build(),\n",
+    "    run_config=RunConfig(\n",
+    "        name=\"vicuna-13b-finetune\",\n",
+    "        storage_path=\"s3://anyscale-staging-data-cld-kvedzwag2qa8i5bjxuevf5i7/air-release-tests\",\n",
+    "        checkpoint_config=CheckpointConfig(\n",
+    "            num_to_keep=1,\n",
+    "            # Enable distributed checkpointing\n",
+    "            _checkpoint_keep_all_ranks=True,\n",
+    "            _checkpoint_upload_from_workers=True,\n",
+    "        ),\n",
+    "    ),\n",
+    "    scaling_config=ScalingConfig(\n",
+    "        num_workers=NUM_WORKERS,\n",
+    "        use_gpu=True,\n",
+    "        resources_per_worker={\"CPU\": 15, \"GPU\": 1},\n",
+    "    ),\n",
+    "    datasets={\"train\": processed_ds},\n",
+    "    datasets_iter_config={\"batch_size\": BATCH_SIZE_PER_WORKER},\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```{tip}\n",
+    "\n",
+    "Here, we highly recommend saving checkpoints with cloud storage and enabling distributed checkpointing by setting `_checkpoint_keep_all_ranks` and `_checkpoint_upload_from_workers` to True when training huge models. Otherwise, all checkpoint shards will be synced to the head node, which may introduce enormous syncing overhead and even cause out-of-memory.\n",
+    "\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model Fine-tuning\n",
+    "\n",
+    "Once everything is configured in LightningTrainer, training becomes easy. Simply call `trainer.fit()`, and your workload will be scaled to the Ray cluster, initiating ZeRO-3 parallel training.\n",
+    "\n",
+    "Now let's do some math here:\n",
+    "\n",
+    "- total training examples = 2379(curated) + 5000(mined) = 7379\n",
+    "- number of workers = 16\n",
+    "- per-worker batch size = 8\n",
+    "- accumulate gradient steps = 2\n",
+    "\n",
+    "Therefore the total effective batches would be `7379 // (16 * 8 * 2) = 29`.\n",
+    "\n",
+    "- Training takes: 36:06 = 2166s\n",
+    "- Training + initialization + checkpointing takes 2473s\n",
+    "\n",
+    "Therefore, the model initialization and checkpoint syncing takes 307s. It will be amortized when you have larger datasets and spend more time on training."
```

Comment:
Move this to after the output, since it's not currently clear where these numbers are coming from.